pageserver/src/tenant/mgr.rs

```diff
@@ -76,6 +79,78 @@ pub async fn init_tenant_mgr(
 
     let mut tenants = HashMap::new();
 
+    // If we are configured to use the control plane API, then it is the source of truth for what to attach
+    let tenant_generations = conf
+        .control_plane_api
+        .as_ref()
+        .map(|control_plane_api| async {
+            let client = reqwest::ClientBuilder::new()
+                .build()
+                .expect("Failed to construct http client");
+
+            // FIXME: it's awkward that join() requires the base to have a trailing slash, makes
+            // it easy to get a config wrong
+            assert!(
+                control_plane_api.as_str().ends_with('/'),
+                "control plane API needs trailing slash"
+            );
```

Comment:
A custom parse function in the config parser is more appropriate than this.
Let's fix it before merging.

Reply:
Fixed this to massage the URL if necessary in ControlPlaneClient, since that's the code that depends on join()'s behavior.

fb3632947a1485b7b5971e6c125a3c8497cbe667